### PR TITLE
split clustering component

### PR DESCRIPTION
**Pull request by Adem-Khachnaoui**

### CODE REVIEW

**Comment by Adem-Khachnaoui** [comment]
test_clustering  will not run anymore. 
I will think of another way to replace it

**Comment by Adem-Khachnaoui** [comment]
@linusmichel can you please change the build workflow.
We need to build the docker for the new embedding component.
and the command :
`python -m unittest discover -s clustering -p test_*.py`
should become 
`python -m unittest discover -s embedding -p test_*.py`

**Comment by linusmichel** [comment]
I adjusted the build workflow including the embedding-tests. One test is failing though, maybe because of a changed model?
Also adjusted the ports according to the logical order of the components

**Comment by linusmichel** [comment]
> I'm getting this error when I try to start components with `docker-compose up -d` command
> `ERROR: Service 'embedding' failed to build: The command '/bin/sh -c pip install --no-cache-dir -qr /tmp/requirements.txt' returned a non-zero code: 137`

@bigultekin try to increase the amount of RAM your Docker-VM is allowed to use. See [Slack](https://ls1tum.slack.com/archives/GUZEEKALS/p1588528912005500) for more infos. @CanArisan encountered this issue before.

**Comment by CanArisan** [comment]
> I'm getting this error when I try to start components with `docker-compose up -d` command
> `ERROR: Service 'embedding' failed to build: The command '/bin/sh -c pip install --no-cache-dir -qr /tmp/requirements.txt' returned a non-zero code: 137`

Yes, as @linusmichel said, increasing the RAM from 2 GB to 4 GB in Docker worked for me.

**Comment by CanArisan** [comment]
> I adjusted the build workflow including the embedding-tests. One test is failing though, maybe because of a changed model?
> I also adjusted the ports according to the logical order of the components.
> @Adem-Khachnaoui don't forget to update the README in the root regarding components and ports.

Which test is failing @linusmichel ? Because test_same_sentences is also failing by me when I try it in PyCharm, but the same happens when I try it in master (only in PyCharm, not in docker). Maybe they are related.

**Comment by linusmichel** [comment]
The failing one is test_distance_permutation (see the [Build-log](https://github.com/ls1intum/Athene/runs/670633659))

**Comment by bigultekin** [comment]
> > I'm getting this error when I try to start components with `docker-compose up -d` command
> > `ERROR: Service 'embedding' failed to build: The command '/bin/sh -c pip install --no-cache-dir -qr /tmp/requirements.txt' returned a non-zero code: 137`
> 
> @bigultekin try to increase the amount of RAM your Docker-VM is allowed to use. See [Slack](https://ls1tum.slack.com/archives/GUZEEKALS/p1588528912005500) for more infos. @CanArisan encountered this issue before.

Well, it took an hour with 8GB :( I guess it is fine since it takes time only in first time

File: docker-compose.yml [hunk 1]
```diff
@@ -11,6 +11,8 @@ services:
     ports:
       - 8000:8000
     working_dir: /usr/src/app
+    volumes:
+      - .:/usr/src/app/logs
```
Comment (linusmichel) [inline docker-compose.yml:15]: This is not what we should do here. This mounts the root-directory of the repo into the /usr/src/app/logs folder.

**Pull request merged**